#!/bin/bash

SPNAME=$1;
SEQUENCE=$2;
GENE=$3;
#echo '$1 = ' $SPNAME ;
#echo '$2 = ' $SEQUENCE ;
#echo '$3 = ' $GENE ;
printf ">$SPNAME\n$SEQUENCE\n" 1>query.fa ;
if [ -e query.fa ]
then
echo -e "query.fa generated\n"
~/bin/tblastn -db ~/mtDB/mtDB -db_gencode 2 -num_alignments 500 -query query.fa -out query.out-mitocode 1>/dev/null 2>/dev/null
if [ -e query.out-mitocode ]
then
echo -e "protein homologs were found\n"
~/bin/mview -in blast -out fasta query.out-mitocode 1>query.out-mitocode.fa 2>/dev/null
if [ -e query.out-mitocode.fa ]
then
~/bin/header_sel_mod3.pl query.out-mitocode.fa $SPNAME 1>query.out-mitocode_sel.fa 2>query.out-mitocode_sel.hash 
printf "fasta protein alignment with $SPNAME and outgroup was generated\nand names legenda was printed (*.namehash)\n"
~/bin/nuc_coding_mod.pl query.out-mitocode_sel.hash 1>query.out-mitocode_sel.nuc 2>/dev/null
if [ -e query.out-mitocode_sel.nuc ]
then
~/bin/codon_alig_unique.pl query.out-mitocode_sel.nuc 1>query.out-mitocode_sel_unique.fa 2>/dev/null
echo -e "\nnucleotide sequences encoding proteins were found\nand doubles were deleted\n"
if [ -e yes.O ]
then
java -jar ~/bin/macse_v1.01b.jar -prog alignSequences -gc_def 2 -out_AA query.out-mitocode_sel_protalign_unique.fa -out_NT query.out-mitocode_sel_tranalign_unique.fa -seq query.out-mitocode_sel_unique.fa
~/bin/macse2.pl query.out-mitocode_sel_tranalign_unique.fa 1>/dev/null 2>/dev/null
~/bin/mutnumbers.pl query.out-mitocode_sel_tranalign_unique.fa 1>$SPNAME.$GENE.mutnumbers 2>/dev/null
echo -e "codon alignment were generated by protein alignment (*.terminals.fa)\nand maximum and minimum mutations number were calculated (*.mutnumbers)\n"
if [ -e query.out-mitocode_sel_tranalign_unique.fa ]
then
java -jar ~/bin/readseq.jar -a -f Phylip -o query.out-mitocode_sel_tranalign.phy query.out-mitocode_sel_tranalign_unique.fa 1>/dev/null 2>/dev/null
~/bin/raxmlHPC-PTHREADS-SSE3 -x 987654 -p 987654 -T 4 -N 50 -f a -s query.out-mitocode_sel_tranalign.phy -n Rax_tree -m GTRGAMMAIX 1>/dev/null 2>/dev/null
echo -e "phylogenetic tree was reconstructed\n"
if [ -e RAxML_bestTree.Rax_tree ]
then
~/bin/newick/bin/nw_distance -m p -s l -n RAxML_bestTree.Rax_tree 1>RAxML_bestTree.Rax_tree.branches 2>/dev/null
sort -grk 2 RAxML_bestTree.Rax_tree.branches 1>RAxML_bestTree.Rax_tree.branches.srt 2>/dev/null
~/bin/newick/bin/nw_reroot -l RAxML_bestTree.Rax_tree OUTGRP 1>RAxML_bestTree.Rax_tree.rooted 2>/dev/null
echo -e "phylogenetic tree was rerooted (*.tree)\nand list of lengths from root was generated(*.tree.srtbrlist)\n";
~/bin/raxmlHPC-PTHREADS-SSE3 -T 4 -f A -m GTRGAMMAIX -s query.out-mitocode_sel_tranalign.phy -t RAxML_bestTree.Rax_tree.rooted -n ANCESTORS 1>/dev/null 2>/dev/null
if [ -e RAxML_nodeLabelledRootedTree.ANCESTORS ]
then
echo -e "ancestral sequences were reconstructed using ML approach (*.MLancestors.fa and *.MLancestors.prob)\nand inner nodes were labelled (*.tree.anclabels)\n"
~/bin/newick/bin/nw_distance -m m -s a -n RAxML_nodeLabelledRootedTree.ANCESTORS 1>RAxML_nodeLabelledRootedTree.ANCESTORS.matrix 2>/dev/null
~/bin/newick/bin/nw_distance -m r -s a -n RAxML_nodeLabelledRootedTree.ANCESTORS 1>RAxML_nodeLabelledRootedTree.ANCESTORS.list2root 2>/dev/null
echo -e "nodal distance from root to any labelled tree node were calculated (*.tree.matrix)\n"
~/bin/pars.pl RAxML_nodeLabelledRootedTree.ANCESTORS query.out-mitocode_sel_tranalign_unique.fa parsimony.tmp.MPancestors.fa 1>/dev/null 2>/dev/null
if [ -e parsimony.tmp.MPancestors.fa ]
then
~/bin/parscompare.pl RAxML_marginalAncestralStates.ANCESTORS parsimony.tmp.MPancestors.fa RAxML_nodeLabelledRootedTree.ANCESTORS.matrix $SPNAME.$GENE.CSancestors.fa 1>$SPNAME.$GENE.CSancestors.MPinc 2>/dev/null
echo -e "ancestral sequences were reconstructed using MP approach (*.MPancestors.fa)\nand compared with the same of ML approach (*.CSancestors.fa - fasta ML/MP incompatibilities masqued by N;\n*.CSancestors.MPinc - listed ML/MP incompatibilities)\n"
mv query.out-mitocode_sel.hash $SPNAME.$GENE.namehash 
mv query.out-mitocode_sel_tranalign_unique.fa $SPNAME.$GENE.terminals.nuc.fa 
mv query.out-mitocode_sel_protalign_unique.fa $SPNAME.$GENE.terminals.pro.fa 
mv RAxML_marginalAncestralProbabilities.ANCESTORS $SPNAME.$GENE.MLancestors.prob 
mv RAxML_marginalAncestralStates.ANCESTORS $SPNAME.$GENE.MLancestors.fa 
mv parsimony.tmp.MPancestors.fa $SPNAME.$GENE.MPancestors.fa 
mv RAxML_nodeLabelledRootedTree.ANCESTORS.matrix $SPNAME.$GENE.tree.matrix 
mv RAxML_nodeLabelledRootedTree.ANCESTORS $SPNAME.$GENE.tree.anclabels 
mv RAxML_bestTree.Rax_tree.rooted $SPNAME.$GENE.tree 
mv RAxML_bestTree.Rax_tree.branches.srt $SPNAME.$GENE.tree.srtbrlist 
#rm query.out-mitocode* yes.O RAxML_* query.fa parsimony.tmp.* outfile outtree
echo -e "files were renamed and temporary files were deleted\n"
fi
fi
fi
fi
else
echo -e "less than 4 unique sequences found - terminating calculation!\n"
#rm query.out-mitocode* query.fa yes.O
fi
fi
fi
fi
fi